opendevin/indexing/repomap/repomap.py

```diff
@@ -0,0 +1,763 @@
+"""
+Mostly borrow from: https://github.com/paul-gauthier/aider/blob/main/aider/repomap.py
+"""
```

Comment:
Make sure you maintain the Apache license at the top of this file attributing the original creator.